### ISSUE TITLE
SCARAB / ChemReg integration

### ISSUE THREAD

**Issue by aekhendry** [thesgc/chembiohub_helpdesk#206]
Based on 18/05/15 meeting requirements, we need to integrate chemreg with SCARAB.

**Comment by aekhendry** [comment]
This is mostly going to be done by David, Really I need to know when he will be able to do this

**Comment by PaulBarrett79** [comment]
Do we ask if a new batch in scarab is created versus a new batch in chemreg and how does that workflow actually look/work?

**Comment by PaulBarrett79** [comment]
We cannot put a time estimate against this until ourr questions to David have been answered.

**Comment by KarenPorter** [comment]
@strets123 please update this issue with current status and anything outstanding for review ahead of sprint review tomorrow.

**Comment by strets123** [comment]
OK, so as scheduled, I have done all of the following tasks:
- [x] Make a separate app to denmonstrate how to override parts of the ChemReg system using class inheritance
- [x] Learn and set up the django url configuration to allow this to work properly
- [x] Refactor the outcomes from the other file upload tasks in the sprint to do with blinded compounds so that we have a simple way to handle sdf files in all the situations required by this integration
- [x] Ensure that configuration files for the uwsgi python server are in place so that the new configuration can be run easily

In an addition to the original plan and to make things easier for David I have also done:
- [x] Create a simple script to install the system on centos including the server dependencies
- [x] Move items where the owner of the anaconda packages had configured them poorly that actually required system level c dependencies back to pip requirements file so that the install on centos proceeds properly

In the time of the sprint it has not been possible to:
[ ] Install and test the system with David

**Comment by strets123** [comment]
Fixes now pushed to David. Icing on cake will be display errors issue but all else is done

**Comment by KarenPorter** [comment]
Is the display errors issue one reported by David? Where he can signal a problem with an individual compound via a specific column, but if the error is more general and applies to all the compounds he doesn't have a way of giving the user a message.  He can throw an exception to block the upload but the user just gets a generic error message.

**Comment by KarenPorter** [comment]
sorry, meant to label to @strets123

**Comment by strets123** [comment]
Currently David has only a way of failing the whole file.

I was going to add line by line failures so that the ignore function can be used but will discuss with David how he wants defaults to be.

For the foreseeable future Scarab and ChemReg batches will just work in their own ways in parallel.

**Comment by aekhendry** [comment]
How do I test this? Is this just waiting for David to test?

**Comment by strets123** [comment]
@ddamerell53 Please comment here when you have done your 2 week test

**Comment by ddamerell53** [comment]
Remind me how to get the username in after_save_and_index_hook

**Comment by strets123** [comment]
request.user.username

**Comment by ddamerell53** [comment]
In the same method how do I access the content of the table?  I need to get
a value from a cell and I'd rather not have to parse the SDF again myself.

**Comment by strets123** [comment]
you will need to do a different request e.g.

        newrequest2 = copy(request)
	newrequest2.GET = request.GET.copy()
        newrequest2.GET["multiple_batch_id"] = multi_batch_id
        newrequest2.GET["offset"] = 0
        newrequest2.GET["limit"] = 10000
        resp = super(AlteredCompoundBatchResource, self).get_list_elasticsearch(newrequest2)

**Comment by ddamerell53** [comment]
What's the format of the object returned by get_list_elasticsearch(). I'd
rather get it correct first time without fiddling around.

**Comment by ddamerell53** [comment]
For simplicity I may as well write the SDF out in Python as well.  Do you
have a request call for getting that as well?

**Comment by strets123** [comment]
Yes, for the SDF just change the format string, copying the request object again

        newrequest = copy(request)
        newrequest.GET = request.GET.copy()
        newrequest.GET["format"] = "sdf"
        newrequest.GET["multiple_batch_id"] = multi_batch_id
        newrequest.GET["offset"] = 0
        newrequest.GET["limit"] = 10000
        file_resp = super(AlteredCompoundBatchResource, self).get_list_elasticsearch(newrequest)

        with open('/tmp/temp.sdf', 'wb') as f:
            f.write(file_resp._container[0])

**Comment by KarenPorter** [comment]
why is this still in backlog list @strets123 ?

**Comment by KarenPorter** [comment]
why is this still in backlog list @strets123 ?

**Comment by KarenPorter** [comment]
Complete. closing
